### DISCUSSION TITLE
Standardized Issues accross Eclipse-Platform repositories

### DISCUSSION THREAD

**Discussion by Wittmaxi** [eclipse-platform/.github#148] [Ideas]
Many [projects on github](https://github.com/audacity/audacity/issues/new/choose) use github's feature to define [issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository)

This may bring a few advantages like having Standardized issues and on average, a Higher quality of issues.

I therefore propose that we implement issue-templates accross our repositories, like [SWT](https://github.com/eclipse-platform/eclipse.platform.swt) already does.

---
We could have the following issue-templates (subject for debate!)

[Bug Report](https://gist.github.com/Wittmaxi/473a4f9f5872b6e402e6d50ffebb46bf)
[Feature Proposal](https://gist.github.com/Wittmaxi/30f37440e769451e8ba0e01a5548a0cd)
[Feature Request](https://gist.github.com/Wittmaxi/c152092a1577e3fd202f13cd15b9e209)
...

**Comment by HeikoKlare** [comment]
I think it is a good idea to have such templates. It does not only (and not even necessarily) unify the way issues look like, but also gives important hints on which kind of information may be important.

I would also propose to automatically assign labels to issues created with different templates, such as a destinction between "bug", "feature request" or "feature proposal".

**Comment by jukzi** [comment]
I am more annoyed by templates. They often ask things that are either obvious, unrelated or unknown.

**Reply by HeikoKlare** [reply]
I understand your point. I see the target group for such templates rather in users of Eclipse RCP/IDE, who may not know what information is important to analyze or even reproduce a problem. This makes it difficult for them to submit an issue and difficult for us to understand a problem. So "both sides" may benefit from templates.

Anyway, I would propose to provide templates only as an offer, not as a requirement. So it would be fine to remove everything that is prefilled or to even provide an empty template for those who do not want to use a template.

**Reply by Wittmaxi** [reply]
I agree, a "blank" template is best for maintainers who already know the workflows

**Reply by mickaelistria** [reply]
I don't think we need nor want a full form (ie a template defined in yaml with widgets and so on), just basic .md files to pre-populate the report provide more flexibility for everyone, from basic users to most experienced ones.
The forms are often very constraining. I don't think we need to put constraints, we just want to put guidance. Some .md files are sufficient for that.

**Reply by mickaelistria** [reply]
I submitted https://github.com/eclipse-platform/.github/pull/171 and plan to merge it soon-ish unless there is major negative feedback.

**Comment by mickaelistria** [comment]
I also support the idea of such templates. They really help the contributor in telling the right things when they're not really aware of what matters to investigate the cause and it doesn't prevent people who feel comfortable enough to from ditching the template if they know a better way to describe their issue.
Templates can also be a good way to set up expectations. In eclipseide-jdtls for instance, we've put a checkbox on the template to make the OSS development more transparent:
"""
- [ ] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

"""
I think template are a very nice way for those who are not much comfortable with the project to bootstrap an efficient technical communication with other contributors, saving roundtrips to gather standard info; and I fail at seeing drawbacks that we can't workaround.

I suggest at this stage, someone can submit a PR to this repo in order to add some concrete issue templates and then we can refine them so they match the expectation of maintainers.
